src/osd/OpRequest.cc

```diff
@@ -82,6 +82,7 @@ void OpRequest::_unregistered() {
   request->clear_data();
   request->clear_payload();
   request->release_message_throttle();
+  request->set_connection(nullptr);
```

Comment:
@gregsfortytwo i re-considered your comment at https://github.com/ceph/ceph/pull/8828#issuecomment-216537466 . and we are safe. see http://tracker.ceph.com/issues/13990#note-105 and https://github.com/ceph/ceph/pull/9108/commits/14207f4390eedb6b59048f07d5467e62fbb6610f .